packages/devkit/src/utils/add-plugin.ts

```diff
@@ -71,27 +86,42 @@ export async function addPlugin<PluginOptions>(
         tree.root,
         nxJson
       );
-    } catch (e) {
-      // Errors are okay for this because we're only running 1 plugin
-      if (e instanceof ProjectConfigurationsError) {
-        projConfigs = e.partialProjectConfigurationsResult;
-      } else {
-        throw e;
-      }
-    }
-    global.NX_GRAPH_CREATION = false;
 
-    for (const projConfig of Object.values(projConfigs.projects)) {
-      const node = graphNodes.find(
-        (node) => node.data.root === projConfig.root
+    const [corePlugins] = await loadNxPlugins([], tree.root);
+    projConfigsWithCorePlugins =
+      await retrieveProjectConfigurationsWithPartialResult(
+        corePlugins,
+        tree.root,
+        nxJson
       );
 
-      if (!node) {
-        continue;
-      }
+    global.NX_GRAPH_CREATION = false;
 
+    for (const projConfig of Object.values(projConfigsWithNewPlugin.projects)) {
       for (const targetName in projConfig.targets) {
-        if (node.data.targets[targetName]) {
+        const existingPluginCreatedTarget =
+          projConfigsWithExistingNxJsonPlugins?.projects?.[projConfig.root]
+            ?.targets?.[targetName];
+
+        const corePluginsCreatedTarget =
+          projConfigsWithCorePlugins?.projects?.[projConfig.root]?.targets?.[
+            targetName
+          ];
+        if (
+          existingPluginCreatedTarget ||
+          (corePluginsCreatedTarget &&
+            isCompatibleTarget(
+              corePluginsCreatedTarget,
+              projConfig.targets[targetName]
+            ) &&
+            !deepEquals(
+              corePluginsCreatedTarget,
+              mergeTargetConfigurations(
+                corePluginsCreatedTarget,
+                projConfig.targets[targetName]
+              )
+            ))
```

Comment:
You don't have to check isCompatible separately from the deep equals imho

```suggestion
          (corePluginsCreatedTarget &&
            !deepEquals(
              corePluginsCreatedTarget,
              mergeTargetConfigurations(
                corePluginsCreatedTarget,
                projConfig.targets[targetName]
              )
            )
```